packages/opentelemetry-core/src/trace/instrumentation/BasePlugin.ts

```diff
@@ -14,15 +14,24 @@
  * limitations under the License.
  */
 
-import { Tracer, Plugin, Logger, PluginConfig } from '@opentelemetry/types';
+import { Tracer, Plugin, Logger, PluginConfig, PluginInternalFiles, PluginInternalFilesVersion } from '@opentelemetry/types';
+import * as semver from 'semver';
+import * as path from 'path';
 
 /** This class represent the base to patch plugin. */
 export abstract class BasePlugin<T> implements Plugin<T> {
+  supportedVersions?: string[];
+  readonly moduleName?: string; // required for internalFilesExports
+  readonly version?: string; // required for internalFilesExports
+  protected readonly _basedir?: string; // required for internalFilesExports
+
   protected _moduleExports!: T;
   protected _tracer!: Tracer;
   protected _logger!: Logger;
+  // tslint:disable-next-line:no-any
+  protected _internalFilesExports!: { [module: string]: any }; // output for internalFilesExports
```

Comment:
could we use `unknown` instead of `any` there ?

Reply:
@vmarchaud With this, the expectation would be to load the types for the internal file in each plugin. Those typings probably probably won't exist if it isn't a typescript library and so the plugin would have to cast it to `any` anyways in that case. I'm okay with this behavior of encouraging trying to use types and passing along the burden of `any` casting to each plugin. WDYT?